### ISSUE TITLE
ReturnCount excludeGuardClauses not working as expected

### ISSUE THREAD

**Issue by Glennmen** [detekt/detekt#5536]
## Expected Behavior

The detekt gradle plugin (and Jetbrains plugin) shouldn't give an error.

## Observed Behavior

> Function initializePreferences has 5 return statements which exceeds the limit of 2. [ReturnCount]

## Steps to Reproduce

The function:

```kotlin
private fun initializePreferences() {
        motionSensorEnabledPreference = findPreference(MOTION_SENSOR_ENABLED) ?: return
        motionSensorTimeoutPreference = findPreference(MOTION_SENSOR_TIMEOUT) ?: return
        isDigitalSignagePreference = findPreference(DIGITAL_SIGNAGE_ENABLED) ?: return
        digitalSignageAppPreference = findPreference(DIGITAL_SIGNAGE_APP_PACKAGE_NAME) ?: return
        digitalSignageTimeoutPreference = findPreference(DIGITAL_SIGNAGE_TIMEOUT) ?: return
        ...
 ```

config/detekt.yml
```yml
style:
    ReturnCount:
        excludeGuardClauses: true
```

## Context

We are currently adding Detekt to our projects and would like to avoid using Suppress.
For our understanding of the rule in combination with `excludeGuardClauses` this case should be marked as correct.

## Your Environment

* Version of detekt used: 1.21.0
* Version of Gradle used (if applicable): 7.0.2
* Gradle scan link (add `--scan` option when running the gradle task): https://scans.gradle.com/s/ye4ux2gb4ae34
* Operating System and version: MacOS 12.6
* Link to your project (if it's a public repository): /
